@artragis Peux-tu regarder le job 3 ? https://travis-ci.org/zestedesavoir/zds-site/jobs/541128526

```
test_no_auto_ping (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_no_dead_notif_on_moving (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_no_dead_ping_notif_on_moving_to_private_forum (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_no_more_notif_on_losing_all_groups (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_no_more_notif_on_losing_one_group (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_no_ping_on_private_forum (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_no_reping_on_edition (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_ping_unknown (zds.notification.tests.tests_tricky.ForumNotification) ... ok
test_no_emails_for_those_who_have_none (zds.notification.tests.tests_tricky.SubscriptionsTest) ... ok
test_no_emails_for_those_who_have_other_things_in_that_place (zds.notification.tests.tests_tricky.SubscriptionsTest) ... ok
test_custom_analyzer (zds.searchv2.tests.tests_models.ESIndexManagerTests)
Test our custom analyzer ... ok
test_indexation (zds.searchv2.tests.tests_models.ESIndexManagerTests)
test the indexation and deletion of the different documents ... ok
test_setup_functions (zds.searchv2.tests.tests_models.ESIndexManagerTests)
Test the behavior of the reset_es_index(), setup_custom_analyzer() and clear_es_index() functions ... ok
test_special_case_of_contents (zds.searchv2.tests.tests_models.ESIndexManagerTests)
test that the old publishedcontent does not stay when a new one is created ... ok
test_es_manager (zds.searchv2.tests.tests_utils.UtilsTests)
Test the behavior of the ``es_manager`` command ... - indexing topics
    1 so far (1000.0 obj/s, batch size: 1000)
  1	items indexed
- indexing posts
    1 so far (2000.0 obj/s, batch size: 2000)
  1	items indexed
- indexing publishedcontents
  2	items indexed
ok
test_basic_search (zds.searchv2.tests.tests_views.ViewsTests)
Basic search and filtering ... ok
test_boosts (zds.searchv2.tests.tests_views.ViewsTests)
Check if boosts are doing their job ... ok
test_category_and_subcategory_impact_search (zds.searchv2.tests.tests_views.ViewsTests)
If two contents do not belong to the same (sub)category ... ok
test_change_publishedcontents_impacts_chapter (zds.searchv2.tests.tests_views.ViewsTests) ... ok
test_change_topic_impacts_posts (zds.searchv2.tests.tests_views.ViewsTests) ... ok
test_get_similar_topics (zds.searchv2.tests.tests_views.ViewsTests)
Get similar topics lists ... ok
test_hidden_forums_give_no_results_if_user_not_allowed (zds.searchv2.tests.tests_views.ViewsTests)
Long name, isn't ? ... ok
test_hidden_post_are_not_result (zds.searchv2.tests.tests_views.ViewsTests)
Hidden posts should not show up in the search results ... ok
test_opensearch (zds.searchv2.tests.tests_views.ViewsTests) ... ok
test_upercase_and_lowercase_search_give_same_results (zds.searchv2.tests.tests_views.ViewsTests)
Pretty self-explanatory function name, isn't it ? ... ok
======================================================================
ERROR: test_failure_initial_content_not_found (zds.featured.tests.FeaturedResourceCreateViewTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/zestedesavoir/zds-site/zds/featured/tests.py", line 205, in test_failure_initial_content_not_found
    .format(reverse('featured-resource-create')))
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/client.py", line 527, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/client.py", line 339, in get
    **extra,
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/client.py", line 414, in generic
    return self.request(**r)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/test/client.py", line 495, in request
    raise exc_value
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/core/handlers/base.py", line 124, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/utils/decorators.py", line 45, in _wrapper
    return bound_method(*args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/utils/decorators.py", line 45, in _wrapper
    return bound_method(*args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/travis/build/zestedesavoir/zds-site/zds/featured/views.py", line 54, in dispatch
    return super(FeaturedResourceCreate, self).dispatch(request, *args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/edit.py", line 168, in get
    return super().get(request, *args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/edit.py", line 133, in get
    return self.render_to_response(self.get_context_data())
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/edit.py", line 66, in get_context_data
    kwargs['form'] = self.get_form()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/edit.py", line 33, in get_form
    return form_class(**self.get_form_kwargs())
  File "/home/travis/build/zestedesavoir/zds-site/zds/featured/views.py", line 95, in get_form_kwargs
    kw = super(FeaturedResourceCreate, self).get_form_kwargs()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/edit.py", line 105, in get_form_kwargs
    kwargs = super().get_form_kwargs()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/django/views/generic/edit.py", line 38, in get_form_kwargs
    'initial': self.get_initial(),
  File "/home/travis/build/zestedesavoir/zds-site/zds/featured/views.py", line 91, in get_initial
    initial.update(**self.get_initial_content_data(content_id))
  File "/home/travis/build/zestedesavoir/zds-site/zds/featured/views.py", line 73, in get_initial_content_data
    displayed_authors = ', '.join([str(x) for x in content.authors.all()])
AttributeError: 'NoneType' object has no attribute 'authors'
----------------------------------------------------------------------
Ran 419 tests in 93.509s
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -69,8 +69,6 @@ cache:
 
 
 before_install:
-  - export HACK_VIRTUALENV=$VIRTUAL_ENV
-
   - source ./scripts/travis_header.sh # Need to be sourced
 
 

--- a/Gulpfile.js
+++ b/Gulpfile.js
@@ -108,13 +108,8 @@ gulp.task('js', () =>
         .pipe(sourcemaps.write('.', { includeContent: true, sourceRoot: '../../' }))
         .pipe(gulp.dest('dist/js/')));
 
-katexFolder = 'zmd'
-if(fs.existsSync('/opt/zmd')) {
-    katexFolder = '/opt/zmd'
-}
-
 gulp.task('prepare-zmd', () =>
-    gulp.src([katexFolder + '/node_modules/katex/dist/{katex.min.css,fonts/*}'])
+    gulp.src(['node_modules/katex/dist/{katex.min.css,fonts/*}'])
         .pipe(gulp.dest('dist/css/')));
 
 // Compiles the SCSS files to CSS

--- a/package.json
+++ b/package.json
@@ -44,6 +44,7 @@
     "gulp.spritesmith": "6.5.1",
     "highlight.js": "^9.12.0",
     "jquery": "3.4.0",
+    "katex": "0.10.1",
     "moment": "^2.20.1",
     "normalize.css": "7.0.0"
   },

--- a/scripts/travis_header.sh
+++ b/scripts/travis_header.sh
@@ -39,13 +39,9 @@ function zds_register_module_for_installation {
 
 
 function install_geckodriver {
-    if [[ "$ZDS_TEST_JOB" == *"selenium"* ]]; then
-        wget "https://github.com/mozilla/geckodriver/releases/download/v0.23.0/geckodriver-v0.23.0-linux64.tar.gz" \
-            && mkdir "geckodriver" \
-            && tar -xzf "geckodriver-v0.23.0-linux64.tar.gz" -C "geckodriver" \
-            && export PATH="$PATH:$PWD/geckodriver" \
-            && export DISPLAY=":99.0"
-    fi
+    wget "https://github.com/mozilla/geckodriver/releases/download/v0.23.0/geckodriver-v0.23.0-linux64.tar.gz"
+    mkdir "geckodriver"
+    tar -xzf "geckodriver-v0.23.0-linux64.tar.gz" -C "geckodriver"
 }
 
 
@@ -102,11 +98,13 @@ if [[ "$ZDS_TEST_JOB" == *"zds.tutorialv2"* ]]; then
         # this script is faster than zds_install.sh +tex-local +latex-template
         git clone $ZDS_LATEX_REPO
         TEMPLATEDIR=$HOME/.texlive/texmf-local/tex/latex/
-        ./latex-template/scripts/install_font.sh \
-        && ./latex-template/scripts/install_texlive.sh \
-        && export PATH=$HOME/.texlive/bin/x86_64-linux:$PATH \
-        && rm -rf $TEMPLATEDIR/latex-template \
-        && mkdir -p $TEMPLATEDIR  && cp -r ./latex-template $TEMPLATEDIR && texhash
+        ./latex-template/scripts/install_font.sh
+        ./latex-template/scripts/install_texlive.sh
+        export PATH=$HOME/.texlive/bin/x86_64-linux:$PATH
+        rm -rf $TEMPLATEDIR/latex-template
+        mkdir -p $TEMPLATEDIR
+        cp -r ./latex-template $TEMPLATEDIR
+        texhash
     zds_fold_end
 fi
 

--- a/scripts/travis_run.sh
+++ b/scripts/travis_run.sh
@@ -35,6 +35,11 @@ function activate_env {
 
 
 activate_env "./$ZDS_VENV"
+	# Add geckodriver in the PATH of zdsenv
+        export PATH="$PATH:/home/travis/build/zestedesavoir/zds-site/geckodriver"
+        export DISPLAY=":99.0"
+	# Use hack for virtualenv (fix some task with "command not found")
+	export PATH="$PATH:/home/travis/virtualenv/python3.5.6/bin"
 
 	run_script "start_elasticsearch"
 
@@ -47,11 +52,6 @@ activate_env "./$ZDS_VENV"
 	run_script "selenium_test"
 
 	run_script "coverage_backend"
-
-
-# Use hack for virtualenv (fix some task with "command not found")
-
-activate_env "$HACK_VIRTUALENV"
 
 	run_script "lint_backend"
 

--- a/scripts/travis_run.sh
+++ b/scripts/travis_run.sh
@@ -46,14 +46,14 @@ activate_env "./$ZDS_VENV"
 
 	run_script "selenium_test"
 
+	run_script "coverage_backend"
+
 
 # Use hack for virtualenv (fix some task with "command not found")
 
 activate_env "$HACK_VIRTUALENV"
 
 	run_script "lint_backend"
-
-	run_script "coverage_backend"
 
 	run_script "build_documentation"
 

--- a/scripts/travis_script.sh
+++ b/scripts/travis_script.sh
@@ -60,6 +60,9 @@ if [[ "$1" == "coverage_backend" ]] && [[ "$ZDS_TEST_JOB" == *"zds."* ]]; then
             exit 1
         fi
 
+        export PATH="$PATH:/home/travis/build/zestedesavoir/zds-site/geckodriver"
+        export DISPLAY=":99.0"
+
         coverage run --source='.' manage.py \
             test -v=2\
             --keepdb \
@@ -68,7 +71,7 @@ if [[ "$1" == "coverage_backend" ]] && [[ "$ZDS_TEST_JOB" == *"zds."* ]]; then
             ${ZDS_TEST_JOB/front/}
 
         #pm2 kill :
-        ./zmd/node_modules/pm2/bin/pm2.js kill; exVal=$?
+        pm2 kill; exVal=$?
 
         if [[ $exVal != 0 ]]; then
             print_error "Warning: Cannot stop zmd"

--- a/scripts/travis_script.sh
+++ b/scripts/travis_script.sh
@@ -60,9 +60,6 @@ if [[ "$1" == "coverage_backend" ]] && [[ "$ZDS_TEST_JOB" == *"zds."* ]]; then
             exit 1
         fi
 
-        export PATH="$PATH:/home/travis/build/zestedesavoir/zds-site/geckodriver"
-        export DISPLAY=":99.0"
-
         coverage run --source='.' manage.py \
             test -v=2\
             --keepdb \

--- a/zds/featured/views.py
+++ b/zds/featured/views.py
@@ -14,7 +14,7 @@ from django.conf import settings
 from zds.featured.forms import FeaturedResourceForm, FeaturedMessageForm
 from zds.featured.models import FeaturedResource, FeaturedMessage
 from zds.forum.models import Topic
-from zds.tutorialv2.models.database import PublishedContent
+from zds.tutorialv2.models.database import PublishableContent
 from zds.utils.paginator import ZdSPagingListView
 
 
@@ -66,8 +66,10 @@ class FeaturedResourceCreate(CreateView):
 
     def get_initial_content_data(self, content_id):
         try:
-            content = PublishedContent.objects.filter(must_redirect=False, content__pk=int(content_id)).first()
-        except (PublishedContent.DoesNotExist, ValueError):
+            content = PublishableContent.objects.get(pk=int(content_id)).public_version
+            if not content:
+                raise ValueError('Not a public content')
+        except (PublishableContent.DoesNotExist, ValueError):
             messages.error(self.request, self.initial_error_message)
             return {}
         displayed_authors = ', '.join([str(x) for x in content.authors.all()])

